### ISSUE TITLE
How to use plugin in shared ESLint config with custom Tailwind classes?

### ISSUE THREAD

**Issue by visualcookie** [francoismassart/eslint-plugin-tailwindcss#435]
### Issue Description

I hope I can use this template for a question, as the discussion section is disabled.

I'm creating a shared ESLint configuration package for our monorepo/organization using **ESLint flat config**. Since all our projects use Tailwind v3 (via Nativewind), I want to include `eslint-plugin-tailwindcss` in this shared config.

**The problem:** Our design system defines custom Tailwind classes. When the plugin is configured directly in each project's `eslint.config.js`, it correctly detects and lints these custom classes. However, when I move the plugin configuration to our shared package (`@company/eslint-config`), it no longer recognizes the custom classes defined in the project's `tailwind.config.js`.

### Current Setup

**Shared config package** (`@company/eslint-config/eslint.config.js`):
```js
const tailwind = require('eslint-plugin-tailwindcss')

module.exports = [
  ...tailwind.configs['flat/recommended'],
  {
    settings: {
      tailwindcss: {
        callees: ['className', 'tv', 'cn'],
        classRegex: '^class(Name)?$',
      },
    },
    rules: {
      'tailwindcss/no-custom-classname': 'error',
      'tailwindcss/classnames-order': 'warn',
      // ... other rules
    },
  },
];
```

**Project's `eslint.config.js`:**
```js
const { defineConfig } = require('eslint/config')
const expoConfig = require('eslint-config-expo/flat')
const projectConfig = require('@company/eslint-config')

module.exports = defineConfig([
  expoConfig,
  ...projectConfig,
])
```

**Project's `tailwind.config.js`** (contains custom classes from our design system)

### Expected Behavior
The plugin should detect and validate custom classes defined in the **consuming project's** `tailwind.config.js`, even when the plugin is configured in a shared package.

### Current Behavior
- ✅ Works: When plugin is configured directly in the project's `eslint.config.js`
- ❌ Doesn't work: When plugin is configured in the shared `@company/eslint-config` package

The plugin appears to not be finding/reading the project's `tailwind.config.js` when used via a shared config.

### Question
Is there a way to configure the plugin in a shared ESLint flat config while still allowing it to resolve the `tailwind.config.js` from the consuming project? I know there is an option parameter for that but I haven't got it working in the shared package yet.

Should I be using the `settings.tailwindcss.config` option differently in flat config? Or is there a way to make the plugin resolve the config relative to the project root rather than the shared package location?

Has anyone successfully used this plugin in a shared ESLint flat configuration with custom Tailwind classes?

Any guidance would be appreciated!

### Environment
- ESLint: 9.x (flat config)
- eslint-plugin-tailwindcss: latest
- Tailwind CSS: v3 (via Nativewind)

**Comment by visualcookie** [comment]
For anyone, who comes across this in later stages as well: Today I continued working on this issue and found out a decent way of handling this properly.

What I did is very simple, initially I try to get the `tailwind.config.js` from the current working directory. If that config is not in the root, but instead in a sub directory, I can now easily add this to my spreading function.

E.g.: `...tailwind('packages/xyz/tailwind.config.js')`

``` cjs
const path = require('path');
const tailwind = require('eslint-plugin-tailwindcss');

const base = [
  {
    rules: {
      '@typescript-eslint/ban-ts-comment': 'warn',
      '@typescript-eslint/no-empty-object-type': 'warn',
      '@typescript-eslint/no-explicit-any': 'warn',
      '@typescript-eslint/no-unused-vars': [
        'warn',
        {
          vars: 'all',
          args: 'after-used',
          ignoreRestSiblings: false,
          argsIgnorePattern: '^_',
          varsIgnorePattern: '^_',
          destructuredArrayIgnorePattern: '^_',
          caughtErrorsIgnorePattern: '^(_|ignore)',
        },
      ],
    },
  },
];

const tailwindConfig = (configPath) => [
  ...tailwind.configs['flat/recommended'],
  {
    settings: {
      tailwindcss: {
        config: configPath ?? path.join(process.cwd(), 'tailwind.config.js'),
        callees: ['className', 'tv', 'cn'],
        classRegex: '^class(Name)?$',
      },
    },
    rules: {
      'tailwindcss/no-custom-classname': 'error',
      'tailwindcss/classnames-order': 'off',
    },
  },
];

module.exports = {
  base,
  tailwind: tailwindConfig,
  expo: (configPath) => [...base, ...tailwindConfig(configPath)],
};
```
